public/app/features/templating/template_srv.ts

```diff
@@ -275,30 +301,31 @@ export class TemplateSrv implements BaseTemplateSrv {
   }
 
   replaceWithText(target: string, scopedVars?: ScopedVars) {
+    deprecationWarning('template_srv.ts', 'replaceWithText()', 'replace(), and specify the :text format');
+
     if (!target) {
       return target;
     }
 
-    let variable;
-    this.regex.lastIndex = 0;
-
-    return target.replace(this.regex, (match: any, var1: any, var2: any, fmt2: any, var3: any) => {
-      if (scopedVars) {
-        const option = scopedVars[var1 || var2 || var3];
-        if (option) {
-          return option.text;
-        }
-      }
+    const targetWithFormat = target.replace(this.regex, (match, var1, var2, fmt2, var3, fieldPath, fmt3) => {
+      const format = fmt2 ?? fmt3;
+      const variable = this.variableNameWithFieldPath(var1 ?? var2 ?? var3, fieldPath);
 
-      variable = this.getVariableAtIndex(var1 || var2 || var3);
       if (!variable) {
         return match;
       }
 
-      const value = this.grafanaVariables[variable.current.value];
+      if (format) {
+        return match.replace(`${variable}:${format}`, `${variable}:text`);
+      }
 
-      return typeof value === 'string' ? value : variable.current.text;
+      if (match.indexOf(`$${variable}`) > -1) {
+        return match.replace(`$${variable}`, `\${${variable}:text}`);
+      }
+      return match.replace(`${variable}`, `${variable}:text`);
     });
```

Comment:
Something about this feels hacky

Reply:
Yeah, it kind of is :)

Reply:
not sure I understand this, I though this function was just to call this.replace(format, scopedVars, 'text');

Reply:
I agree with @torkelo, we should just call this.replace(format, scopedVars, 'text') here instead?

Reply:
ahh, I didn't see the forest for all the trees. Thanks 🙏